agents/src/voice/audio_recognition.ts

```
@@ -37,7 +37,9 @@ export class AudioRecognition {
 
   start() {
     this.vadStreamProcessor = this.vadTask().catch((err) => {
-      this.logger.error('Error in VAD task', err);
+      // raise the error
+      throw err;
+      this.logger.error(`Error in VAD task: ${err}`);
     });
```

What's the proper way to handle errors from these async functions we run in node? I'll do some digging on this. Ideally we want to raise them rather than swallow them.